tests/integration/test_recipes_loading.py

```diff
@@ -130,12 +143,13 @@ def which(executable):
         extract_shape = preproc.get('extract_shape')
         if extract_shape and 'shapefile' in extract_shape:
             filename = Path(
-                config_user['auxiliary_data_dir']) / extract_shape['shapefile']
+                session['auxiliary_data_dir']) / extract_shape['shapefile']
             filename.parent.mkdir(parents=True, exist_ok=True)
             filename.touch()
 
     # Account for module change after esmvalcore=2.7
     if version.parse(core_ver) <= version.parse('2.7.1'):
+        config_user = session.to_config_user()
         esmvalcore._recipe.read_recipe_file(recipe_file, config_user)
     else:
-        esmvalcore._recipe.recipe.read_recipe_file(recipe_file, config_user)
+        esmvalcore._recipe.recipe.read_recipe_file(recipe_file, session)
```

Comment:
ah! Was bashing my head on Friday why this wasn't picking up the correct structure :beer:

Reply:
It needs an `esmvalcore.config.Session` object now instead of a  plain `dict`.